sw/device/tests/penetrationtests/firmware/crypto_fi.c

```diff
@@ -0,0 +1,282 @@
+    .message_len = 200,
+    .customization_string = "My Tagged Application",
+    .customization_string_len = 21,
+    .digest = {0x1c73bed5, 0x73d74e95, 0x59bb4628, 0xe3a8e3db, 0x7ae7830f,
+               0x5944ff4b, 0xb4c2f1f2, 0xceb8ebec, 0xc601ba67, 0x57b88a2e,
+               0x9b492d8d, 0x6727bbd1, 0x90117868, 0x6a300a02, 0x1d28de97,
+               0x5d3030cc},
+    .digest_len = 16,
+    .digest_len_is_fixed = false,
+};
+
+static const uint8_t kKeyShare1[] = {
+    0x0f, 0x1f, 0x2f, 0x3f, 0x4f, 0x5f, 0x6f, 0x7f, 0x8f, 0x9f, 0xaf,
+    0xbf, 0xcf, 0xdf, 0xef, 0xff, 0x0a, 0x1a, 0x2a, 0x3a, 0x4a, 0x5a,
+    0x6a, 0x7a, 0x8a, 0x9a, 0xaa, 0xba, 0xca, 0xda, 0xea, 0xfa,
+};
+
+static dif_aes_transaction_t transaction = {
+    .operation = kDifAesOperationEncrypt,
+    .mode = kDifAesModeEcb,
+    .key_len = kDifAesKey128,
+    .manual_operation = kDifAesManualOperationManual,
+    .key_provider = kDifAesKeySoftwareProvided,
+    .mask_reseeding = kDifAesReseedPer8kBlock,
+    .reseed_on_key_change = false,
+    .force_masks = false,
+    .ctrl_aux_lock = false,
+};
+
+status_t handle_crypto_fi_aes(ujson_t *uj) {
+  // Get the test mode.
+  crypto_fi_aes_mode_t uj_data;
+  TRY(ujson_deserialize_crypto_fi_aes_mode_t(uj, &uj_data));
+  // Clear registered alerts in alert handler.
+  uint32_t reg_alerts = sca_get_triggered_alerts();
+
+  // Write the key into the AES block. Set and unset the trigger when
+  // key_trigger is true.
+  if (uj_data.key_trigger) {
+    sca_set_trigger_high();
+  }
+  TRY(dif_aes_start(&aes, &transaction, &aes_key_shares, NULL));
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusInputReady, true, TIMEOUT);
+  if (uj_data.key_trigger) {
+    sca_set_trigger_low();
+  }
+
+  // Write the plaintext into the AES block. Set and unset the trigger when
+  // plaintext_trigger is true.
+  if (uj_data.plaintext_trigger) {
+    sca_set_trigger_high();
+  }
+  TRY(dif_aes_load_data(&aes, aes_plaintext));
+  if (uj_data.plaintext_trigger) {
+    sca_set_trigger_low();
+  }
+
+  // Start the encryption. Set and unset the trigger when encrypt_trigger is
+  // true.
+  if (uj_data.encrypt_trigger) {
+    sca_set_trigger_high();
+  }
+  TRY(dif_aes_trigger(&aes, kDifAesTriggerStart));
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusOutputValid, true, TIMEOUT);
+  if (uj_data.encrypt_trigger) {
+    sca_set_trigger_low();
+  }
+
+  // Read the ciphertext. Set and unset the trigger when ciphertext_trigger is
+  // true.
+  dif_aes_data_t ciphertext;
+  if (uj_data.ciphertext_trigger) {
+    sca_set_trigger_high();
+  }
+  TRY(dif_aes_read_output(&aes, &ciphertext));
+  if (uj_data.ciphertext_trigger) {
+    sca_set_trigger_low();
+  }
+
+  // Get registered alerts from alert handler.
+  reg_alerts = sca_get_triggered_alerts();
+
+  // Send the ciphertext and the alerts back to the host.
+  crypto_fi_aes_ciphertext_t uj_output;
+  memcpy(uj_output.ciphertext, (uint8_t *)ciphertext.data, 16);
```

Comment:
You don't need the cast here do you?
```suggestion
  memcpy(uj_output.ciphertext, ciphertext.data, 16);
```

Reply:
Agree, thanks for pointing this out.